src/upload.ts

```diff
@@ -154,34 +153,32 @@ async function uploadVideo(videoJSON: Video, messageTransport: MessageTransport)
             if (progressChecker == undefined || !curProgress) return
             curProgress = curProgress.split(" ").find((txt: string) => txt.indexOf("%") != -1)
             let newProgress = curProgress ? parseInt(curProgress.slice(0, -1)) : 0
-            if ( progress.progress == newProgress ) return
+            if (progress.progress == newProgress) return
             progress.progress = newProgress
             videoJSON.onProgress!(progress)
         }, 500)
     }
 
     const errorMessage = await page.evaluate(() => (document.querySelector('.error-area.style-scope.ytcp-uploads-dialog') as HTMLElement)?.innerText.trim())
-
     if (errorMessage) {
         await browser.close()
         throw new Error('Youtube returned an error : ' + errorMessage)
     }
 
     // Wait for upload to complete
-    const uploadCompletePromise = page.waitForXPath('//*[contains(text(),"Upload complete")]', { timeout: 0 }).then(() => 'uploadComplete')
+    const uploadCompletePromise = page.waitForXPath('//tp-yt-paper-progress[contains(@class,"ytcp-video-upload-progress-hover") and @value="100"]', { timeout: 0 }).then(() => 'uploadComplete')
```

Comment:
Why are we changing this selector ?
Can't see any connection with any of the listed features in the merge request description.
Has this been tested with both Shorts and non-shorts video ?

Reply:
I change it because of if you upload the same video twice the second will be instantly success despite daily limit 
Currently , not testest yet with short video...